lucille-core/src/test/java/com/kmwllc/lucille/core/RunnerTest.java

```diff
@@ -755,32 +755,30 @@ public void testIndexerDeleteMarkerConfig() throws Exception {
     assertEquals(0, SharedMetricRegistries.names().size());
-    Runner.runInTestMode("RunnerTest/twoConnectors.conf");
+    Map<String, TestMessenger> runnerMessengerMap = Runner.runInTestMode("RunnerTest/twoConnectors.conf");
+    String connector1ID = runnerMessengerMap.get("connector1").getRunId();
+    String connector2ID = runnerMessengerMap.get("connector2").getRunId();
     assertEquals(1, SharedMetricRegistries.names().size());
     assertEquals(LogUtils.METRICS_REG, SharedMetricRegistries.names().toArray()[0]);
     MetricRegistry metrics = SharedMetricRegistries.getOrCreate(LogUtils.METRICS_REG);
 
     // confirm that each stage instance within each connector/pipeline pair creates a timer with a unique name
     // note: it's important for names to be unique, otherwise metrics across instances of a stage would be combined
     SortedMap<String, Timer> connector1Pipeline1Stage1Timers =
-        metrics.getTimers(MetricFilter.contains("connector1.pipeline1.stage.stage_1.processDocumentTime"));
+        metrics.getTimers(MetricFilter.contains(connector1ID + ".connector1.pipeline1.stage.stage_1.processDocumentTime"));
     SortedMap<String, Timer> connector2Pipeline2Stage1Timers =
-        metrics.getTimers(MetricFilter.contains("connector2.pipeline2.stage.stage_1.processDocumentTime"));
+        metrics.getTimers(MetricFilter.contains(connector2ID + ".connector2.pipeline2.stage.stage_1.processDocumentTime"));
     assertEquals(1, connector1Pipeline1Stage1Timers.size());
     assertEquals(1, connector2Pipeline2Stage1Timers.size());
 
     // each timer should have a count of one because each connector produces one document
     assertEquals(1, connector1Pipeline1Stage1Timers.get(connector1Pipeline1Stage1Timers.firstKey()).getCount());
     assertEquals(1, connector2Pipeline2Stage1Timers.get(connector2Pipeline2Stage1Timers.firstKey()).getCount());
 
-    // currently, metrics for the same connector/pipeline pairs will be combined across mulitple runs occurring in the
-    // same JVM;
-    // if we re-execute the run from above the counts will not be reset but will increase from their earlier values;
-    // we may wish to prevent this in the future by including the runId in the metrics naming scheme
-    // or by updating Runner to call SharedMetricRegistries.clear() before each run, assuming runs are sequential
+    // metrics for the same connector/pipeline pairs are independent across multiple runs in the same JVM.
     Runner.runInTestMode("RunnerTest/twoConnectors.conf");
-    assertEquals(2, connector1Pipeline1Stage1Timers.get(connector1Pipeline1Stage1Timers.firstKey()).getCount());
-    assertEquals(2, connector2Pipeline2Stage1Timers.get(connector2Pipeline2Stage1Timers.firstKey()).getCount());
+    assertEquals(1, connector1Pipeline1Stage1Timers.get(connector1Pipeline1Stage1Timers.firstKey()).getCount());
+    assertEquals(1, connector2Pipeline2Stage1Timers.get(connector2Pipeline2Stage1Timers.firstKey()).getCount());
```

Comment:
the variables connector1Pipeline1Stage1Timers and connector2Pipeline2Stage1Timers should be renamed to run1connector1Pipeline1Stage1Timers and run1connector2Pipeline2Stage1Timers

the current assertions should be kept in place -- this is showing that the metrics for run1 are unchanged by the execution of run2

but we also need to capture run2connector1Pipeline1Stage1Timers and run2connector2Pipeline2Stage1Timers and assert that those are set correctly